surfacers/cloudwatch/cloudwatch.go

```diff
@@ -50,7 +47,45 @@ type CWSurfacer struct {
 
 	// A cache of []types.MetricDatum's, used for batch writing to the
 	// cloudwatch api.
-	cwMetricDatumCache []types.MetricDatum
+	metricDatumCache       []types.MetricDatum
+	metricDatumCacheTicker *time.Ticker
+}
+
+// New creates a new instance of a cloudwatch surfacer, based on the config
+// passed in. It then hands off to a goroutine to surface metrics to cloudwatch
+// across a buffered channel.
+func New(ctx context.Context, conf *configpb.SurfacerConf, opts *options.Options, l *logger.Logger) (*CWSurfacer, error) {
+	region := getRegion(conf)
+
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(region))
+	if err != nil {
+		return nil, err
+	}
+
+	cw := &CWSurfacer{
+		c:                      conf,
+		opts:                   opts,
+		writeChan:              make(chan *metrics.EventMetrics, opts.Config.GetMetricsBufferSize()), // incoming internal metrics buffer
+		session:                cloudwatch.NewFromConfig(cfg),
+		l:                      l,
+		metricDatumCache:       make([]types.MetricDatum, 0, int(conf.GetMetricsBatchSize())), // batching buffer between cloudprober and cloudwatch
+		metricDatumCacheTicker: time.NewTicker(time.Duration(conf.GetMetricsPublishTimerSec()) * time.Second),
+	}
+
+	go cw.processIncomingMetrics(ctx)
+
+	cw.l.Infof("Initialised Cloudwatch surfacer with batchsize: %d, publish timer (secs): %d\n", conf.GetMetricsBatchSize(), conf.GetMetricsPublishTimerSec())
+	return cw, nil
+}
+
+// Write is a function defined to comply with the surfacer interface, and enables the
+// cloudwatch surfacer to receive EventMetrics over the buffered channel.
+func (cw *CWSurfacer) Write(ctx context.Context, em *metrics.EventMetrics) {
+	select {
+	case cw.writeChan <- em:
+	default:
+		cw.l.Error("Surfacer's write channel is full, dropping new data.")
+	}
 }
 
 func (cw *CWSurfacer) processIncomingMetrics(ctx context.Context) {
```

Comment:
Regarding my comment above about moving ticker creation here:
```suggestion
func (cw *CWSurfacer) processIncomingMetrics(ctx context.Context) {
    batchTimeTicker := time.NewTicker(...)
    defer batchTimeTicker.Stop()
```